tests/unit/core/scan/test_scan_context.py

```diff
@@ -114,7 +114,8 @@ def test_ci_no_env(env, fake_url_repo: Repository) -> None:
     WHEN there is no repository url in the environment
     THEN the remote url is sent by default
     """
-    with mock.patch.dict(os.environ, env, clear=True):
+    environ = {**os.environ.copy(), **env}
```

Comment:
If I am not mistaken, not reusing the existing environment is done on purpose: to ensure the `GITHUB_ACTIONS` environment variable is not set when running on CI. Why did you have to make this change?

Reply:
This tests failed for windows because it couldn't find git. This test seemingly only succeeded before because it found the git path in the cache.

Reply:
I modified it to remove the GITHUB_ACTIONS env var

Reply:
Wow, side-effects everywhere!

Odd that GITHUB_ACTIONS is not set. According to the environment dump from my ci-sandbox repository (https://github.com/agateau-gg/ci-sandbox/actions/runs/7491395928/job/20392507364), it should be there.

There is an autouse fixture in tests/unit/cmd/scan/test_ci.py to remove it, but I would have expected it to not run outside of its module? 🤔

Reply:
I spent a bit of time looking at this and I think I understand what is happening: this test checks that when the CI env variables do not contain the necessary info, it falls back to getting the repository URL in some other way, and this way is independent from the CI type.

I think the `GITHUB_ACTIONS` env var is set but it does not change anything *visible*.

Here is a test I did:

I add this code inside the `with` block:

```python
        from ggshield.core.git_hooks.ci import SupportedCI
        print(f"SupportedCI={SupportedCI.from_ci_env()}")
```

and defined `GITHUB_ACTIONS` to `1` in my environment.

I then ran the test (using `-vs` to see the output) before and after your changes.

Before:

```
tests/unit/core/scan/test_scan_context.py::test_ci_no_env[env0] SupportedCI=SupportedCI.AZURE
PASSED
tests/unit/core/scan/test_scan_context.py::test_ci_no_env[env1] SupportedCI=SupportedCI.DRONE
PASSED
tests/unit/core/scan/test_scan_context.py::test_ci_no_env[env2] SupportedCI=SupportedCI.GITHUB
PASSED
tests/unit/core/scan/test_scan_context.py::test_ci_no_env[env3] SupportedCI=SupportedCI.GITLAB
PASSED
tests/unit/core/scan/test_scan_context.py::test_ci_no_env[env4] SupportedCI=SupportedCI.CIRCLECI
PASSED
tests/unit/core/scan/test_scan_context.py::test_ci_no_env[env5] SupportedCI=SupportedCI.BITBUCKET
PASSED
```

After:

```
tests/unit/core/scan/test_scan_context.py::test_ci_no_env[env0] SupportedCI=SupportedCI.GITHUB
PASSED
tests/unit/core/scan/test_scan_context.py::test_ci_no_env[env1] SupportedCI=SupportedCI.GITHUB
PASSED
tests/unit/core/scan/test_scan_context.py::test_ci_no_env[env2] SupportedCI=SupportedCI.GITHUB
PASSED
tests/unit/core/scan/test_scan_context.py::test_ci_no_env[env3] SupportedCI=SupportedCI.GITLAB
PASSED
tests/unit/core/scan/test_scan_context.py::test_ci_no_env[env4] SupportedCI=SupportedCI.GITHUB
PASSED
tests/unit/core/scan/test_scan_context.py::test_ci_no_env[env5] SupportedCI=SupportedCI.GITHUB
PASSED
```

As you can see, the app does not think it is in the same environment. This looks wrong. Can you try by only copying the $PATH environment variable?